lib/instrumentation/generic-span.js

```diff
@@ -101,3 +120,16 @@ GenericSpan.prototype.addLabels = function (labels) {
   }
   return true
 }
+
+GenericSpan.prototype.setType = function (type, subtype, action) {
+  if (type && type.indexOf('.') >= 0) {
```

Comment:
If we're making this a breaking change, we don't have to support the old format anymore as far as I can see. If so, we don't need the custom `Object.defineProperty` setter either, which also simplifies things 😃

Reply:
Hard-dropping support for setting it with the old format could break existing custom instrumentation though. I think it's probably best to put it through a deprecation cycle. 🤔